scripts/install-kubefed.sh

```diff
@@ -2,79 +2,94 @@
-    i) IMAGE_NAME=$OPTARG ;;
+  i) IMAGE_NAME=$OPTARG ;;
+  s) SCOPE=$OPTARG;;
     esac
 done
 
 echo "NS=$NAMESPACE"
 echo "LOC=$LOCATION"
 echo "Operator Image Name=$IMAGE_NAME"
+echo "Scope=$SCOPE"
 
-if test X"$NAMESPACE" != X; then
+if test X"$NAMESPACE" != Xdefault; then
     # create a namespace 
     kubectl create ns ${NAMESPACE}
-    NAMESPACE_STR="--namespace=${NAMESPACE}"
 fi
 
-# Install CRD
+# Install kubefed CRD
 kubectl apply -f ./deploy/crds/operator_v1alpha1_kubefed_crd.yaml
 
-# Install CR
-kubectl apply -f ./deploy/crds/operator_v1alpha1_kubefed_cr.yaml $NAMESPACE_STR
+# Install kubefed CR based on the scope
+if test X"$SCOPE" = XCluster; then
+  sed "s,scope:.*,scope: ${SCOPE}," ./deploy/crds/operator_v1alpha1_kubefed_cr.yaml | kubectl apply -n $NAMESPACE -f -
+else
+  kubectl apply -f ./deploy/crds/operator_v1alpha1_kubefed_cr.yaml -n $NAMESPACE
+fi
+
 
-# Check if operator-sdk is installed or not and accordinlgy execute the command.
+# A local deployment
 if test X"$LOCATION" = Xlocal; then
   operator-sdk &> /dev/null
   if [ $? == 0 ]; then
   # operator-sdk up local command doesn't install the requried CRD's
   for f in ./deploy/crds/*_crd.yaml ; do     
 	  kubectl apply -f "${f}" --validate=false 
   done
-	    operator-sdk up local $NAMESPACE_STR &
+	    operator-sdk up local --namespace=$NAMESPACE &
   else
 	    echo "Operator SDK is not installed."
 	    exit 1
   fi
+
+# in-cluster deployment on kind cluster
 elif test X"$LOCATION" = Xcluster; then
-  #TODO: change the location in the container stanza of the operator yaml
   for f in ./deploy/*.yaml ; do
    if test X"$OPERATOR_YAML_PATH" = X"$f" ; then
       echo "Reading the image name and sed it in"
-      sed "/image: /s|: .*|: ${IMAGE_NAME}|" $f | kubectl apply $NAMESPACE_STR --validate=false -f -
-    else
-      kubectl apply -f "${f}" --validate=false $NAMESPACE_STR
-    fi
+      sed "/image: /s|: .*|: ${IMAGE_NAME}|" $f | kubectl apply -n $NAMESPACE --validate=false -f -
+   elif test X"$CLUSTER_ROLEBINDING" = X"$f" ; then
+      echo "Reading the namespace in clusterrolebinding and sed it in"
+      sed "/namespace: /s|: .*|: ${NAMESPACE}|" $f | kubectl -n $NAMESPACE apply -f -
+   else
+      kubectl apply -f "${f}" --validate=false -n $NAMESPACE
+   fi
   done
   for f in ./deploy/crds/*_crd.yaml ; do     
 	  kubectl apply -f "${f}" --validate=false 
   done
   echo "Deployed all the operator yamls for kubefed-operator in the cluster"
 
+# olm-deployment on minikube cluster
 elif test X"$LOCATION" = Xolm-kube; then
-./scripts/kubernetes/olm-install.sh ${OLM_VERSION}
+ ./scripts/kubernetes/olm-install.sh ${OLM_VERSION}
 
-    echo "OLM is deployed in the cluster"
- 
-    ./hack/catalog.sh | kubectl apply $NAMESPACE_STR -f -
+ echo "OLM is deployed in the cluster"
+ #TODO look for a better approach to avoid in-place sed command
+ sed "s,image: quay.*$,image: ${IMAGE_NAME}," -i $CSV_PATH|./hack/catalog.sh | kubectl apply -n $NAMESPACE -f -
```

Comment:
why do you need to do -i $CSV_PATH? if you take that out, this will be effectively in-place without the need to clobber the csv file. Look for how it's done on Line71

Reply:
`sed "/image: /s|: .*|: ${IMAGE_NAME}|" $f | kubectl apply -n $NAMESPACE --validate=false -f -`
you're passing the above sed command as an input to `kubectl` but that's not how it's handled in this case. You need to change the content of this (./deploy/olm-catalog/kubefed-operator/0.1.0/kubefed-operator.v0.1.0.clusterserviceversion.yaml) file which in turn will be utilized by the `catalog.sh` script.
`CSV=$(cat $(find $DIR/deploy/olm-catalog -name '*version.yaml' | sort -n) | indent apiVersion)` Let me know if that makes sense to you.

Reply:
@abhat let me know if you've any feedback or suggestions here.